### ISSUE TITLE
After a sucessful auth request seeing "Exception of type 'System.AggregateException' was thrown." when calling a cloud function.

### ISSUE THREAD

**Issue by Rainking80** [firebase/quickstart-unity#270]
Hi,
My app successfully finishes an auth request (running in editor). User is logged in correctly. Then I call a cloud function and it finishes with "Exception of type 'System.AggregateException' was thrown."

What I do:

1.  FirebaseApp.CheckAndFixDependenciesAsync().ContinueWith(task => { 
       firebaseApp = FirebaseApp.DefaultInstance;
       ...
2.  later: auth = Firebase.Auth.FirebaseAuth.GetAuth(firebaseApp);
3. in AuthStateChanged then creating global "functions" object: functions = FirebaseFunctions.GetInstance(firebaseApp);
4. calling the function exactly as described here: https://firebase.google.com/docs/functions/callable 
5. Every function call ends with task.IsFaulted = true and empty stack trace
6. this is the error message: "Exception of type 'System.AggregateException' was thrown."

additional informations:
- app is correctly signed (I see the problem in editor AND when deployed to android device)
- everything runs in "us-entral1"
- using the lates googleplaygames plugin (0.9.56) and firebase_unity_sdk_5.4.3 (just the packages auth, remoteConfig, database and functions)

I am searching since 3 days for the root cause, I guess now I need some help...
Thank you!
Regards,
Tino

**Comment by bklimt** [comment]
Hi @Rainking80,

Have you looked at the backend logs to see what the Functions are doing? You can find them in the Firebase console, in Functions, in the Logs tab. They should say something like:

`Function execution took 7 ms, finished with status code: 200`

If they instead show a red flag and some error text, then there's some unhandled exception in your backend function. If not, you may want to add some logging statements to your functions just to make sure they're doing what you expect.

On the client, all exceptions are wrapped in `AggregateException`, but there may be a more specific exception you can get out of it. An example of how to do that is here:

https://github.com/firebase/quickstart-unity/blob/fe63718246942a212554e8a0a46fb76140753b3b/functions/testapp/Assets/Firebase/Sample/Functions/TestCase.cs#L63

**Comment by Rainking80** [comment]
Hi,

Thank you for the fast response. In my firebase console I see the functions successfully called (attached screenshot). Also I use Postman to develop and test the functions - everything seems to run perfectly except the Unity client:

![image](https://user-images.githubusercontent.com/5574367/49859034-d315bf00-fdf6-11e8-9b1e-ac4c2c7660e4.png)

**Comment by bklimt** [comment]
Hmm, and if you unwrap the `AggregateException` object, what do you see?

        FunctionsException e = null;
        foreach (var inner in task.Exception.InnerExceptions) {
          if (inner is FunctionsException) {
            e = (FunctionsException)inner;
            break;
          }
        }
        if (e == null) {
          // We didn't get a proper Functions Exception.
          throw task.Exception;
        }
        throw new Exception(String.Format("Error {0}: {1}", e.ErrorCode, e.Message));

**Comment by Rainking80** [comment]
Hi,

thank you! with your hint and the help of the debugger I found the root cause. It was a type mismatch when returning the result. Since I took the exact code from the Firebase documentation here: https://firebase.google.com/docs/functions/callable 
I guess it is too old. Other code samples like this one are working: https://github.com/firebase/quickstart-unity/blob/fe63718246942a212554e8a0a46fb76140753b3b/functions/testapp/Assets/Firebase/Sample/Functions/TestCase.cs 

Kind regards,
Tino
